diff --git a/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt b/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt
--- a/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt
+++ b/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt
@@ -199,7 +199,8 @@
             if (status == BluetoothGatt.GATT_SUCCESS) {
                 resolve(key, "Characteristic successfully written.")
             } else {
-                reject(key, "Writing characteristic failed.")
+                val statusHex = String.format("0x%02X", status)
+                reject(key, "Writing characteristic failed. status=$status ($statusHex)")
             }
 
         }
